pkg/job/delete_all.go

```diff
@@ -0,0 +1,49 @@
+package job
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "mayadata.io/d-operators/types/job"
+	"openebs.io/metac/dynamic/clientset"
+)
+
+func (r *TaskRunner) deleteAll() (*types.TaskStatus, error) {
+	var message = fmt.Sprintf(
+		"Delete: Resource %s %s: GVK %s",
+		r.Task.DeleteAll.State.GetNamespace(),
+		r.Task.DeleteAll.State.GetAPIVersion(),
+		r.Task.DeleteAll.State.GetKind(),
+	)
+	var client *clientset.ResourceClient
+	var err error
+	err = r.Retry.Waitf(
+		func() (bool, error) {
+			client, err = r.GetClientForAPIVersionAndKind(
+				r.Task.DeleteAll.State.GetAPIVersion(),
+				r.Task.DeleteAll.State.GetKind(),
+			)
+			if err != nil {
+				return r.IsFailFastOnDiscoveryError(), err
+			}
+			return true, nil
+		},
+		message,
+	)
+	if err != nil {
+		return nil, err
+	}
+	err = client.
+		Namespace(r.Task.DeleteAll.State.GetNamespace()).
+		Delete(
+			r.Task.DeleteAll.State.GetKind(),
```

Comment:
Delete does not take kind as an argument. It takes Name.
In this case we should use DeleteAll or DeleteCollection method.

Reply:
Thanks :+1: